libpod/runtime.go

```diff
@@ -328,6 +330,16 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (retErr error) {
 
 	runtime.mergeDBConfig(dbConfig)
 
+	unified, _ := cgroups.IsCgroup2UnifiedMode()
+	if unified && rootless.IsRootless() && !systemd.IsSystemdSessionValid(rootless.GetRootlessUID()) {
+		// If user is rootless and XDG_RUNTIME_DIR is found, podman will not proceed with /tmp directory
+		// it will try to use existing XDG_RUNTIME_DIR
+		// if current user has no write access to XDG_RUNTIME_DIR we will fail later
+		if unix.Access(runtime.storageConfig.RunRoot, unix.W_OK) != nil {
```

Comment:
I am not sure this is the right place for the check.

`runtime.storageConfig.RunRoot` could have been set for a different reason than `XDG_RUNTIME_DIR` being specified.

Would it work if we have it in `SetXdgDirs` for the case where `os.Getenv("XDG_RUNTIME_DIR") != ""` ?

Reply:
@giuseppe first version of PR had check exactly where you suggested, moved it after @mheon suggestion  reason is added in the comment itself https://github.com/containers/podman/pull/11327#issuecomment-906401764

Reply:
This seems more sane - there's no guarantee Podman is actually using XDG_RUNTIME_DIR (we can fall back to other directories). This guarantees that we are checking whatever the currently-configured temporary files directory is.

Reply:
thanks for the clarification.  It makes sense.